app/app/users/page.tsx

```diff
@@ -0,0 +1,97 @@
+import Card from "react-bootstrap/Card";
+import Spinner from "react-bootstrap/Spinner";
+import Table from "react-bootstrap/Table";
+import AppBreadCrumb from "@/components/AppBreadCrumb";
+import { useUsersInfinite } from "@/utils/users";
+import { useToken } from "@/utils/auth";
+import AlignedLabel from "@/components/AlignedLabel";
+import { FaUserPlus, FaCopy } from "react-icons/fa6";
+
+export default function Users() {
+  const token = useToken();
+  const router = useRouter();
+  const { users, isLoading, error } = useUsersInfinite(token);
+
+  if (error) {
+    console.error(error);
+  }
+
+  return (
+    <>
+      <AppBreadCrumb />
+      <Card>
+        <Card.Header className="fs-5 fw-bold">Users</Card.Header>
+        <Card.Body>
+          <div className="mb-3">
+            {!isLoading && (
+              <>
+                <Button className="me-2">
+                  <AlignedLabel>
+                    <FaUserPlus className="me-2" />
+                    <span>Add user</span>
+                  </AlignedLabel>
+                </Button>
+                <Button>
+                  {/* todo: https://github.com/cityofaustin/atd-data-tech/issues/20121 */}
+                  <AlignedLabel>
+                    <FaCopy className="me-2" />
+                    <span>Copy user emails - todo</span>
+                  </AlignedLabel>
+                </Button>
+              </>
+            )}
+          </div>
+          {/* todo: standardize the way we show error messages and use error boundary */}
+          {Boolean(error) && (
+            <Alert variant="dange">Something went wrong</Alert>
+          )}
+          <Table responsive hover>
+            <thead>
+              <tr>
+                <th>Name</th>
+                <th>Email</th>
+                <th>Created at</th>
+                <th>Last login</th>
+                <th>Login count</th>
+                <th>Role</th>
+              </tr>
+            </thead>
+            <tbody>
+              {isLoading && (
+                <tr>
+                  <td colSpan={6} className="text-center">
+                    <Spinner variant="primary" />
+                  </td>
+                </tr>
+              )}
+              {users.map((user) => {
+                return (
+                  <tr
+                    key={user.user_id}
+                    style={{ cursor: "pointer" }}
+                    onClick={() => router.push(`/users/${user.user_id}`)}
+                  >
+                    <td>{user.name}</td>
+                    <td>{user.email}</td>
+                    <td>{new Date(user.created_at).toLocaleDateString()}</td>
+                    <td>
+                      {user.last_login
+                        ? new Date(user.last_login).toLocaleDateString()
+                        : ""}
+                    </td>
+                    <td>{user.logins_count}</td>
+                    <td>{user.app_metadata.roles[0]}</td>
```

Comment:
This is putting like `readonly`, `vz-admin`, `editor` and `itSupervisor` which isn't wrong, but these values feel like `code` to me and not human role names. Do you think it would be helpful to make a lookup object to turn these names into "Viewer", "Administrator", "Editor" and "System Administrator"?

Reply:
Absolutely—I will make a little utility function for that and push up that change 🙏